cli/sketch/new.go

```diff
@@ -47,17 +48,21 @@ void loop() {
 `)
 
 func runNewCommand(cmd *cobra.Command, args []string) {
-	sketchDir := globals.Config.SketchbookDir.Join(args[0])
-	if err := sketchDir.MkdirAll(); err != nil {
+	sketchDir, err := filepath.Abs(args[0])
+	if err != nil {
+		feedback.Errorf("Error creating sketch: %v", err)
+		os.Exit(errorcodes.ErrGeneric)
+	}
+	if err := os.MkdirAll(sketchDir, os.FileMode(0755)); err != nil {
 		feedback.Errorf("Could not create sketch directory: %v", err)
 		os.Exit(errorcodes.ErrGeneric)
 	}
-
-	sketchFile := sketchDir.Join(args[0] + ".ino")
-	if err := sketchFile.WriteFile(emptySketch); err != nil {
+	sketchName := filepath.Base(sketchDir)
+	sketchFile := filepath.Join(sketchDir, sketchName+".ino")
```

Comment:
Not sure if worth it, but we might strip the `.ino` suffix if already present in `args[0]` (e.g. maybe an user might do `sketch new foo.ino`)

Reply:
Good point, added a `TrimSuffix` upstream.